### ISSUE TITLE
ld.lld: error: undefined symbol: __asan::FlushUnneededASanShadowMemory(unsigned int, unsigned int)

### ISSUE THREAD

**Issue by hiraditya** [llvm/llvm-project#107468]
Happened with one of the recent patches, I haven't bisected it but putting the error log here.

```
ld.lld: error: undefined symbol: __asan::FlushUnneededASanShadowMemory(unsigned int, unsigned int)
>>> referenced by asan_allocator.cpp:309 ([out/llvm-project/compiler-rt/lib/asan/asan_allocator.cpp:309])
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_allocator.cpp.o:(__sanitizer::LargeMmapAllocator<__asan::AsanMapUnmapCallback, __sanitizer::LargeMmapAllocatorPtrArrayStatic, __sanitizer::LocalAddressSpaceView>::Deallocate(__sanitizer::AllocatorStats*, void*))
>>> referenced by asan_fake_stack.cpp:76 ([out/llvm-project/compiler-rt/lib/asan/asan_fake_stack.cpp:76]
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_fake_stack.cpp.o:(__asan::FakeStack::Destroy(int))

ld.lld: error: undefined symbol: __asan::AsanApplyToGlobals(void (*)(__asan_global*, unsigned int), void const*)
>>> referenced by asan_globals.cpp:396 ([out/llvm-project/compiler-rt/lib/asan/asan_globals.cpp:396])
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_globals.cpp.o:(__asan_register_image_globals)
>>> referenced by asan_globals.cpp:404 ([out/llvm-project/compiler-rt/lib/asan/asan_globals.cpp:404])
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_globals.cpp.o:(__asan_unregister_image_globals)

ld.lld: error: undefined symbol: __sanitizer::MapDynamicShadow(unsigned long, unsigned long, unsigned long, unsigned long&, unsigned long)
>>> referenced by asan_linux.cpp:95 ([out/llvm-project/compiler-rt/lib/asan/asan_linux.cpp:95]
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_linux.cpp.o:(__asan::FindDynamicShadowStart())

ld.lld: error: undefined symbol: __sanitizer::UnmapFromTo(unsigned long, unsigned long)
>>> referenced by asan_linux.cpp:83 ([out/llvm-project/compiler-rt/lib/asan/asan_linux.cpp:83]
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_linux.cpp.o:(__asan::FindDynamicShadowStart())

ld.lld: error: undefined symbol: __asan::UnpoisonStack(unsigned long, unsigned long, char const*)
>>> referenced by asan_posix.cpp:55 ([out/llvm-project/compiler-rt/lib/asan/asan_posix.cpp:55]
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_posix.cpp.o:(__asan::PlatformUnpoisonStacks())
>>> referenced by asan_posix.cpp:65 ([out/llvm-project/compiler-rt/lib/asan/asan_posix.cpp:65]
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_posix.cpp.o:(__asan::PlatformUnpoisonStacks())

ld.lld: error: undefined symbol: __sanitizer::GetThreadStackAndTls(bool, unsigned long*, unsigned long*, unsigned long*, unsigned long*)
>>> referenced by asan_posix.cpp:63 ([out/llvm-project/compiler-rt/lib/asan/asan_posix.cpp:63]
>>>               lib/asan/CMakeFiles/RTAsan_dynamic.arm.dir/asan_posix.cpp.o:(__asan::PlatformUnpoisonStacks())
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by kongy** [comment]
Bisection points to commit 4d819daab91f54b90365927ba4b40e5a2eff26a9.

**Comment by kongy** [comment]
This was fixed by commit db7e8f2ae81fe10170dc202e45ee8b784e75c74c
